cfgov/ask_cfpb/migration_scripts/migrate_knowledgebase.py

```diff
@@ -0,0 +1,281 @@
+from __future__ import unicode_literals
+
+import time
+
+from knowledgebase.models import QuestionCategory as QC
+from knowledgebase.models import Question, Audience, UpsellItem, EnglishAnswer
+from ask_cfpb.models import (
+    Answer,
+    # AnswerPage,
+    Category,
+    NextStep,
+    SubCategory)
+from ask_cfpb.models import Audience as ASK_audience
+
+
+def get_kb_statuses(ask_id):
+    q = Question.objects.get(id=ask_id)
+    return {'en': (q.get_english_answer_state() == 'APPROVED'),
+            'es': (q.get_spanish_answer_state() == 'APPROVED')}
+
+
+def create_answer_pages(queryset):
+    counter = 0
+    if queryset:
+        for answer in queryset:
+            kb_statuses = get_kb_statuses(answer.id)
+            for lang in ['en', 'es']:
+                if kb_statuses[lang]:
+                    page = answer.create_or_update_page(language=lang)
+                    revision = page.get_latest_revision()
+                    revision.publish()
```

Comment:
I don't think this is actually an issue here. The hook only catches "after create" and "after edit_page" actions, which are admin page transactions, and doesn't touch `revision.publish()` itself. 
So the migration publish step, same as Wagtail's `publish_scheduled_pages` management  command used once a minute in our Jenkins `cf.gov-wagtail-publish-pages-cron-jobs` job, doesn't fire the sharing hook.

I tested this by setting a pdb trace before the akamai flush. If I edit and publish a page, the trace steps in. If I set a go-live datetime and run `publish_scheduled_pages` after the scheduled time, the page publishes and no trace is set.